_overviews/scala3-migration/scala3-migrate.md

```diff
@@ -0,0 +1,315 @@
+
+```scala
+//build.sbt
+lazy val main = project
+  .in(file("."))
+  .settings(
+    name := "main",
+    scalaVersion := "2.13.7",
+    semanticdbEnabled := true,
+    scalacOptions ++= (if (scalaVersion.value.startsWith("3")) Seq("-Werror", "-Ykind-projector")
+    else Seq("-Werror", "-Wunused")),
+    libraryDependencies ++= (
+      if (scalaVersion.value.startsWith("3")) Seq()
+      else
+        Seq(compilerPlugin("org.typelevel" %% "kind-projector" % "0.13.2" cross CrossVersion.full))
+      ),
+    libraryDependencies ++= Seq(
+      "org.typelevel"                    %% "cats-core"           % "2.6.1",
+      "ch.epfl.scala"                     % "scalafix-interfaces" % "0.9.26",
+      "ch.epfl.scala" %% "scalafix-rules" % "0.9.26" % "test" cross CrossVersion.for3Use2_13
+    )
+  )
+```
+
+## Migrate `scalacOptions`
+**Usage:** `migrate-scalacOptions projectId` where projectId is the name of the module chosen to be migrated.
+
+This command helps with the process of updating the compiler settings. It is based on 
+[the Compiler Options Table](options-lookup.html).
+Between Scala 2.13 and Scala 3.1.0, the available compiler options are different:
+- some Scala 2.13 settings have been removed
+- others have been renamed 
+- some remain the same.
+
+The previous build file specifies two scalacOptions: `-Werror` and `-Wunused`
+```
+> migrate-scalacOptions main
+[info] X       : the option is not available in Scala 3
+[info] Renamed : the option has been renamed
+[info] Valid   : the option is still valid
+[info] Plugin  : the option is related to a plugin, previously handled by migrate-libs
+[info]
+[info] -Wunused -> X
+[info] -Werror  -> -Xfatal-warnings
+
+[info] Plugins options
+[info] -Yrangepos -> X
+[info] -Xplugin:/Users/meriamlachkar/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/kind-projector_2.13.3/0.13.2/kind-projector_2.13.3-0.13.2.jar       -> Plugin
+[info] -Xplugin:/Users/meriamlachkar/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalameta/semanticdb-scalac_2.13.3/4.4.20/semanticdb-scalac_2.13.3-4.4.20.jar -> Plugin
+[info] -P:semanticdb:synthetics:on                                                                                                                                             -> Plugin
+[info] -P:semanticdb:sourceroot:/Users/meriamlachkar/perso/plugin-test                                                                                                         -> Plugin
+[info] -P:semanticdb:targetroot:/Users/meriamlachkar/perso/plugin-test/target/scala-2.13/meta                                                                                  -> Plugin
+[info] -P:semanticdb:failures:warning                                                                                                                                          -> Plugin
+
+```
```

Comment:
I think we should explain why we see things like `-P:semanticdb:failures:warning`, although the build file specifies only two options, `-Werror` and `-Wunused`.

Reply:
it's explained in the section below (I have added `and all the specific options of the `semanticdb` plugin starting by `-P:semanticdb:...``  to be even clearer.
```
 - `Xplugin:semanticdb` and all specific options of semanticdb plugin starting by `-P:semanticdb:...` are 
 added through an sbt setting `semanticdbEnabled := true`
that is set by scala3-migrate (this tool). If `semanticdb` is added through `compilerPlugin` or 
`addCompilerPlugin`, it will be listed as a library dependency when we execute migrate-libs. 
````

Reply:
Is that resolved this point ?